### ISSUE TITLE
USB keyboard periodically freezes 

### ISSUE THREAD

**Issue by jimryan** [tekezo/Karabiner-Elements#619]
I haven't yet figured out what triggers this, but my keyboard periodically (maybe once an hour or so) stops responding for 10-20 seconds.  If it happens to freeze while I have a key pressed, that key is repeated while this event is occurring.  Either way, the keyboard is completely unresponsive while it's frozen.  It always begins working again within a few seconds without me doing anything.  I've used this same keyboard with other computers without issue, and I've used this same keyboard with this same computer for a while without issue, which leads me to believe that this is due to a recent Karabiner update.  Whenever it locks up, I've noticed this in the logs:

> [2017-02-11 15:25:04.617] [error] [grabber] IOHIDDeviceSetValue error -536870186 for USB Keyboard @ ___ZN22human_interface_device23set_caps_lock_led_stateEN4krbn9led_stateE_block_invoke

I'm using a Filco FILCKTL15 on a late 2016 13" MacBook Pro running macOS 10.12.3.  Karabiner Elements 0.90.86.

**Comment by jimryan** [comment]
Also, I forgot to mention that my keypresses are still captured and eventually processed while my keyboard is frozen.

**Comment by glyph** [comment]
I started seeing this issue recently as well (in fact much more frequently than every hour; it would happen almost instantly when I launched the app); after trying to roll back to a few previous versions, what eventually seems to have addressed it for me is uninstalling Karabiner-Elements and re-installing it.  This seemed to have happened shortly after the update to 0.90.86 but simply reinstalling earlier versions didn't help, nor did simply rebooting.  Weird!

**Comment by jimryan** [comment]
Thank you!  I've done that and will report back.

**Comment by jimryan** [comment]
This still appears to be an issue for me :(

**Comment by gusutabopb** [comment]
I think this is the same issue as in #593.

**Comment by jimryan** [comment]
Yes, I think you are right - I missed that issue in my search.  Oddly enough, it doesn't impact my system as much, though.  Everything else works fine while the keyboard is frozen.  My mouse works, applications are responsive, etc.  Even the keys pressed while the keyboard is frozen are eventually sent to whatever application has focus when the keyboard unlocks.

**Comment by veselyslon** [comment]
last week i have the same problem with wireless logitech k270:
`[2017-02-21 13:02:34.917] [error] [grabber] IOHIDDeviceOpen error: -536870206 @ ___ZN22human_interface_device4grabEv_block_invoke_2
[2017-02-21 13:02:34.917] [error] [grabber] IOHIDDeviceSetValue error -536870195 for Unifying Device @ ___ZN22human_interface_device23set_caps_lock_led_stateEN4krbn9led_stateE_block_invoke
[2017-02-21 13:02:34.917] [error] [grabber] IOHIDDeviceSetValue error -536870195 for Unifying Device @ ___ZN22human_interface_device23set_caps_lock_led_stateEN4krbn9led_stateE_block_invoke
[2017-02-21 13:02:34.917] [error] [grabber] IOHIDDeviceSetValue error -536870195 for Keyboard @ ___ZN22human_interface_device23set_caps_lock_led_stateEN4krbn9led_stateE_block_invoke
[2017-02-21 13:02:34.917] [error] [grabber] IOHIDDeviceSetValue error -536870195 for Keyboard @ ___ZN22human_interface_device23set_caps_lock_led_stateEN4krbn9led_stateE_block_invoke
[2017-02-21 13:02:34.917] [error] [grabber] IOHIDDeviceSetValue error -536870195 for USB Receiver @ ___ZN22human_interface_device23set_caps_lock_led_stateEN4krbn9led_stateE_block_invoke
[2017-02-21 13:02:34.917] [error] [grabber] IOHIDDeviceSetValue error -536870195 for Keyboard @ ___ZN22human_interface_device23set_caps_lock_led_stateEN4krbn9led_stateE_block_invoke
[2017-02-21 13:02:34.917] [error] [grabber] IOHIDDeviceOpen error: -536870206 @ ___ZN22human_interface_device4grabEv_block_invoke_2`

**Comment by rshmiraf** [comment]
I thought my keyboard is broken. I've tried with 3 keyboard and everyone have freezes every about 5 minutes. Please fix it :)

**Comment by veselyslon** [comment]
A new solution?

**Comment by letalumil** [comment]
I experience the same issue with my keyboard:
```
[2017-06-23 14:15:46.489] [error] [grabber] IOHIDDeviceSetValue error -536870186 for USB Keyboard (device_id:3) @ ___ZN4krbn22human_interface_device23set_caps_lock_led_stateENS_9led_stateE_block_invoke
[2017-06-23 14:25:15.469] [error] [grabber] IOHIDDeviceSetValue error -536870186 for USB Keyboard (device_id:3) @ ___ZN4krbn22human_interface_device23set_caps_lock_led_stateENS_9led_stateE_block_invoke
[2017-06-23 14:39:03.423] [error] [grabber] IOHIDDeviceSetValue error -536870186 for USB Keyboard (device_id:3) @ ___ZN4krbn22human_interface_device23set_caps_lock_led_stateENS_9led_stateE_block_invoke
```

**Comment by damiencollard** [comment]
Same issue here, Filco Majestouch-2 TKL.

**Comment by caglar10ur** [comment]
Same problem here with 0.91.13 using Whitefox Aria

```
[2017-08-16 15:48:54.816] [error] [grabber] IOHIDDeviceSetValue error -536850432 for Keyboard - WhiteFox PartialMap pjrcUSB full (device_id:5) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke
[2017-08-16 15:48:54.816] [error] [grabber] IOHIDDeviceSetValue error -536850432 for Keyboard - WhiteFox PartialMap pjrcUSB full (device_id:3) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke
[2017-08-16 15:48:55.877] [error] [grabber] IOHIDDeviceSetValue error -536850432 for Keyboard - WhiteFox PartialMap pjrcUSB full (device_id:5) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke
[2017-08-16 15:48:55.878] [error] [grabber] IOHIDDeviceSetValue error -536850432 for Keyboard - WhiteFox PartialMap pjrcUSB full (device_id:3) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke
```

**Comment by jagarciavi** [comment]
Same issue, I'm using version 0.91.13

`[2017-08-20 18:27:50.370] [error] [grabber] IOHIDDeviceSetValue error -536870206 for USB Receiver (device_id:14) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke
[2017-08-20 18:27:50.381] [error] [grabber] IOHIDDeviceSetValue error -536870206 for USB Receiver (device_id:12) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke
[2017-08-20 18:27:50.381] [error] [grabber] IOHIDDeviceSetValue error -536870206 for USB Receiver (device_id:13) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke`

**Comment by amneacsu** [comment]
Same issue for me too. Filco Majestouch 2.

**Comment by chriscarpenter12** [comment]
Experiencing the same issue with my Filco Majestouch 2

**Comment by hgl** [comment]
Same issue with Filco Majestouch 2

**Comment by ivanalejandro0** [comment]
I'm experiencing almost the same. 2-5 seconds freeze.
I'm using a CODE keyboard.

**Comment by lzhoucs** [comment]
Same issue. I am on macos 10.12.6 (16G29) with Karabiner-Elements version_11.1.0

**Comment by drummingdemon** [comment]
+1 , WASD v2

``[2017-11-02 17:46:28.659] [error] [grabber] IOHIDDeviceSetValue error -536870186 for USB Keyboard (device_id:3) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke
[2017-11-02 17:46:32.945] [info] [grabber] Re-enable event_tap_ by kCGEventTapDisabledByTimeout
[2017-11-02 18:32:03.611] [error] [grabber] IOHIDDeviceSetValue error -536870186 for USB Keyboard (device_id:3) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke
[2017-11-02 18:33:39.563] [error] [grabber] IOHIDDeviceSetValue error -536870186 for USB Keyboard (device_id:3) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke
[2017-11-02 18:33:39.567] [info] [grabber] Re-enable event_tap_ by kCGEventTapDisabledByTimeout
[2017-11-02 18:40:54.503] [error] [grabber] IOHIDDeviceSetValue error -536870186 for USB Keyboard (device_id:3) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke
[2017-11-02 18:40:54.641] [info] [grabber] Re-enable event_tap_ by kCGEventTapDisabledByTimeout
[2017-11-02 19:00:32.533] [error] [grabber] IOHIDDeviceSetValue error -536870186 for USB Keyboard (device_id:3) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke``

**Comment by liub3** [comment]
Poker 2 with latest version 
[2017-11-06 17:06:46.126] [error] [grabber] IOHIDDeviceSetValue error -536854449 for USB Receiver (device_id:2) @ IOReturn krbn::human_interface_device::set_caps_lock_led_state(krbn::led_state)_block_invoke

**Comment by shinmiy** [comment]
Fixed in 11.2.0?
https://github.com/tekezo/Karabiner-Elements/blob/master/NEWS.md
> The caps lock LED manipulation has been disabled with non Apple keyboards until it is enabled manually.

**Comment by jimryan** [comment]
It looks like that did fix the issue - but at the cost of my caps lock indicator.  It's worth giving up the LED if it means my keyboard won't lock up, but something still feels wrong if I have to make that decision.
